src/__checks__/table-index.check.ts

```diff
@@ -0,0 +1,47 @@
+import { defineDb, defineTable, integer, text, timestampWithTimeZone, uuid } from '..';
+import { expect, describe, test } from 'tstyche';
+import { btree, gin, gist } from '../table-index-types';
+import { Index } from '../table-index';
+
+const foo = defineTable({
+  columns: {
+    id: uuid().primaryKey().default(`gen_random_uuid()`),
+    createDate: timestampWithTimeZone().notNull().default(`now()`),
+    name: text().notNull(),
+    value: integer(),
+  },
+  defineIndexes: (foo) => ({
+    fooPkey: btree(foo.id).primaryKey(),
+    fooCompound: btree(foo.id, foo.name).unique(),
+    fooGist: gist(foo.id, foo.createDate),
+    fooGin: gin(foo.name, foo.value),
+    fooCovering: btree(foo.id).unique().include(foo.name, foo.value, foo.createDate),
+    fooNonNull: btree(foo.id).where(foo.value.isNotNull()),
+    fooExpression: gin(foo.id, foo.name, foo.value.gt(25)).where(foo.name.eq('foo')),
+  }),
+});
+
+const db = defineDb({ foo }, () => Promise.resolve({ rows: [], affectedCount: 0 }));
+const indexes = db.foo.getIndexes();
+
+describe('defineIndexes', () => {
+  test('should preserve index names as type keys', () => {
```

Comment:
Why is it important to preserve the index names as types?

Reply:
It's not that important - in fact, none of the `Index` generic type parameters are important, since unlike columns, we won't use any indexes in queries and write code that needs to know information about the resulting types. They're nice-to-haves, but they're not that nice. Do you want me to try and drop them and see how much simpler things get?

Reply:
Yeah, I think we should drop them. That keeps the `Table` type with only 2 type parameters.

Reply:
I'll leave the `Indexes` in the `Table` type for now until we talk on Friday, if we want to leave them in to make future changes to check query efficiency easier.